### PR TITLE
Hexid

### PR DESCRIPTION
**Pull request by montjoie**
This PR fix the problem of malformated udev lines via two methods:
- prevent the use of string in idvendor/idproduct
- print the ID via "%04x"

This PR is based on top of enablekvm PR
This PR support also python3 for lavalab-gen.py

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ So you have now:
       type: beaglebone-black
       uart:
         idvendor: "0403"
-	idproduct: 6001
+	idproduct: "6001"
```
Comment (khilman) [inline README.md]: No, please do not require quotes.  Instead requre (and document in the examples) to use a hex prefix 0x.

File: boards.yaml [hunk 2]
```diff
@@ -1,4 +1,66 @@
...
   boardlist:
     qemu-01:
       type: qemu
+    bcm2837-rpi-3-b-01:
+      type: bcm2837-rpi-3-b
+      pdu_generic:
+        hard_reset_command: /usr/local/bin/acme-cli -s 192.168.66.2 reset 1
+        power_off_command: /usr/local/bin/acme-cli -s 192.168.66.2 switch_off 1
+        power_on_command: /usr/local/bin/acme-cli -s 192.168.66.2 switch_on 1
+      uart:
```
Comment (khilman) [inline boards.yaml]: The adding of all these boards should not be part of the master branch.  This should be in a lab-specific branch someplace.

File: lavalab-gen.py [hunk 3]
```diff
@@ -92,6 +92,10 @@ def main(args):
                 baud = b["uart"].get("baud", baud_default)
                 idvendor = b["uart"]["idvendor"]
                 idproduct = b["uart"]["idproduct"]
+                # Prevent that convertion of the hex value
+                if type(idproduct) == int or type(idvendor) == int:
+                    print("Please encapsulate IDs for %s with quotes" % board_name)
+                    sys.exit(1)
```
Comment (khilman) [inline lavalab-gen.py]: Instead, read the values as hex (which will valditate they are hex) and then print them in udev rules as hex with `%04x`

**Review by khilman — CHANGES_REQUESTED** [review]

File: boards.yaml (hunk 2 above)
Comment (montjoie) [reply boards.yaml]: This should never be in this PR, I have commited this by mistake sorry.

File: lavalab-gen.py (hunk 3 above)
Comment (montjoie) [reply lavalab-gen.py]: print as hex is done, but we cannot read as hex, only as int so we cannot know if the value was with prefixed wit 0x or not

**Review by khilman — APPROVED** [review]

**Pull request merged**